pandas/core/generic.py

```diff
@@ -4418,7 +4418,7 @@ def add_prefix(self: NDFrameT, prefix: str) -> NDFrameT:
         # expected "NDFrameT")
         # error: Argument 1 to "rename" of "NDFrame" has incompatible type
         # "**Dict[str, partial[str]]"; expected "Union[str, int, None]"
-        return self.rename(**mapper)  # type: ignore[return-value, arg-type]
+        return self.rename(**mapper)
```

Comment:
ah, this needs to be `self._rename` (and then you'll likely need to put the `type: ignore` back - sorry, I hadn't spotted this earlier